controllers/scoring.py

```diff
@@ -0,0 +1,183 @@
+        if self.gripper.game_piece_in_reach():
+            self.gripper.close()
+            if self.gripper.get_full_closed():
+                self.is_holding = field.GamePiece.CUBE
+                self.next_state("idle")
+
+    @state
+    def auto_pickup_cube(self):
+        """Drives to the single substation to intake a cube"""
+        if not self.autodrive:
+            self.next_state("idle")
+
+        self.movement.set_goal(*self.get_cube_pickup())
+        if self.movement.time_to_goal < 2:
+            self.arm.goto_setpoint(Setpoints.HANDOFF)
+            self.gripper.open()
+            self.intake.deploy()
+            if self.gripper.game_piece_in_reach():
+                self.gripper.close()
+                self.is_holding = field.GamePiece.CUBE
+            if self.gripper.get_full_closed():
+                self.next_state("idle")
+        elif self.movement.time_to_goal > 5:
+            self.arm.goto_setpoint(Setpoints.STOW)
+        self.movement.do_autodrive()
+
+    @state
+    def auto_pickup_cone(self):
+        if not self.autodrive:
+            self.next_state("idle")
+
+        self.movement.set_goal(*self.get_cone_pickup())
+        self.intake.retract()
+        if self.movement.time_to_goal < self.grab_pre_time:  # or gripper sees cone
+            self.gripper.close()
+            if self.gripper.get_full_closed():
+                self.is_holding = field.GamePiece.CONE
+                self.next_state("idle")
+        elif self.movement.time_to_goal < 3:
+            self.arm.goto_setpoint(Setpoints.PICKUP_CONE)
+            self.gripper.open()
+        elif self.movement.time_to_goal > 5:
+            self.arm.goto_setpoint(Setpoints.STOW)
+
+        self.movement.do_autodrive()
+
+    @state
+    def score(self, initial_call):
+        if not self.autodrive:
+            self.next_state("idle")
+
+        if initial_call:
+            move_goal, self.arm_setpoint = self.get_score_location()
+            self.movement.set_goal(*move_goal)
+
+        self.intake.retract()
+        if self.movement.time_to_goal < 0.1:
+            self.gripper.open()
+            self.is_holding = field.GamePiece.NONE
+            if self.gripper.get_full_open():
+                self.next_state("idle")
+        elif self.movement.time_to_goal < 4:
+            self.arm.goto_setpoint(self.arm_setpoint)
+        elif self.movement.time_to_goal > 5:
+            self.arm.goto_setpoint(Setpoints.STOW)
+
+        self.movement.do_autodrive()
+
+    @feedback
+    def is_red(self) -> bool:
+        return self.get_team() == wpilib.DriverStation.Alliance.kRed
+
+    def get_team(self) -> wpilib.DriverStation.Alliance:
+        return wpilib.DriverStation.getAlliance()
+
+    def get_current_piece(self) -> field.GamePiece:
+        """What piece the gripper is currently holding"""
+        if self.gripper.get_full_closed() or self.gripper.is_opening():
+            return self.is_holding
+        return field.GamePiece.NONE
```

Comment:
Is this only going to be defined by in the gripper? I might be taking this out of context but if we have a cube in the well after intaking then this shouldnt return none right?